scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -5,6 +5,7 @@
 import numpy as np
 from numpy.testing import assert_allclose
 from pytest import raises as assert_raises
+pytest.importorskip("scipy.sparse.linalg._svdp")
```

Comment:
This change should not be here either.

Reply:
Now there are no changes for this file, though I'm a bit confused--it seems to me like a runtime skip guard on the env variable for the module is still warranted here, similar to `pytestmark = pytest.mark.skipif(not os.environ.get(..`

Reply:
Probably not needed since each individual test already skips correctly and the import will always work, but it doesn't matter much either way.

Reply:
@rgommers @h-vetinari Are we sure about this? The 1.8.0rc3 OSX failures over at conda-forge in https://github.com/conda-forge/scipy-feedstock/pull/198 have tests suites that go like this:

```
...
sparse/linalg/tests/test_propack.py::test_svdp[LM-True-single-array] PASSED [ 63%]
sparse/linalg/tests/test_propack.py::test_svdp[LM-True-single-csr_matrix] PASSED [ 63%]
sparse/linalg/tests/test_propack.py::test_svdp[LM-True-single-csc_matrix] PASSED [ 63%]
sparse/linalg/tests/test_propack.py::test_svdp[LM-True-double-array] PASSED [ 63%]
sparse/linalg/tests/test_propack.py::test_svdp[LM-True-double-csr_matrix] PASSED [ 63%]
sparse/linalg/tests/test_propack.py::test_svdp[LM-True-double-csc_matrix] PASSED [ 63%]
Fatal Python error: Segmentation fault

Current thread 0x000000010f49cdc0 (most recent call first):

```

Shouldn't those be skipping when the environment variable is not set? That's what worried me here--I didn't understand/see how these particular tests were autoskipping without some patch on our end.